iwyu_driver.cc

```diff
@@ -274,6 +274,21 @@ bool ExecuteAction(int argc, const char** argv,
     args.push_back("-Qunused-arguments");
   }
 
+#if defined(IWYU_CLANG_EXECUTABLE_PATH)
+  // Use the full Clang executable path provided by CMake to compute Clang's
+  // resource dir, which is the base for the compiler builtin headers (stddef.h
+  // and friends). Add an explicit -resource-dir argument to tell IWYU to look
+  // there.
+  std::string resource_dir =
+      Driver::GetResourcesPath(IWYU_CLANG_EXECUTABLE_PATH);
```

Comment:
On Fedora, this function is returning `/usr/lib64/clang/18`.
The correct path is `/usr/lib/clang/18`.

I can't tell if this is a bug upstream or downstream.
So, I reported https://bugzilla.redhat.com/show_bug.cgi?id=2299949 for further investigation.

Reply:
Thank you!

Reply:
I've just found more information about this.
I'm afraid the second argument of `Driver::GetResourcesPath()` is important, otherwise it doesn't provide the correct path all the time.
I believe this should be written as `Driver::GetResourcesPath(IWYU_CLANG_EXECUTABLE_PATH, CLANG_RESOURCE_DIR)`.

See an example here:
https://github.com/llvm/llvm-project/blob/main/clang/lib/Driver/Driver.cpp#L242

And I'm afraid that moves us back to the original issue: does `include-what-you-use` get the value of `CLANG_RESOURCE_DIR`.

Reply:
@tuliom Thanks! I saw that `CLANG_RESOURCE_DIR` argument in the constructor, but didn't quite think about the ramifications.Hmm. I don't know enough about packaging on various platforms to suggest what a better behavior might be, we'll see if something falls out in the upstream discussions.